addons/l10n_id/models/account_move.py

```diff
@@ -0,0 +1,152 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import pytz
+
+from odoo import api, fields, models, _
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    # Used to fetch the QR payment status.
+    l10n_id_qris_invoice_details = fields.Json(
+        string="QRIS Transaction Number",
+        help="Transaction Number stored.",
+        readonly=True,
+        export_string_translation=False,
+    )
+    # Used to return the same QR code if still valid.
+    l10n_id_latest_qris_content = fields.Char(
+        string="QRIS Content",
+        help="QRIS Content stored as a char.",
+        readonly=True,
+        export_string_translation=False,
+    )
+
+    def _generate_qr_code(self, silent_errors=False):
+        """
+        Adds information about which invoice is triggering the creation of the QR-Code, so that we can link both together.
+        """
+        self.env.context = {**self.env.context, "qris_originating_invoice_id": self.id}
+        return super()._generate_qr_code(silent_errors)
+
+    def _cron_update_payment_status(self):
+        """
+        This cron will:
+            - Get all invoices that are not paid, and have details about QRIS qr codes.
+            - For each invoices, get information about the payment state of the QR using the API.
+            - If the QR is not paid and it has been more than 30m, we discard that qr id (no longer valid)
+            - If it is paid, we will register the payment on the invoices.
+        """
+        invoices = self.search([
+            ('payment_state', '=', 'not_paid'),
+            ('l10n_id_qris_invoice_details', '!=', False)
+        ])
+        return invoices._update_payment_status()
+
+    def action_update_payment_status(self):
+        """ Do the same thing as the cron, but triggered manually. If payments are created, they will be displayed. """
```

Comment:
Not very docstring-like :wink: 
If I see this in my IDE or in the documentation, I won't get far without inspecting the code...